graylog2-web-interface/src/views/components/TimerangeForForm.ts

```diff
@@ -22,8 +22,21 @@ import type { TimeRange } from 'views/logic/queries/Query';
 import type { SearchBarFormValues } from 'views/Constants';
 import { isTypeRelativeWithStartOnly, isTypeRelativeWithEnd } from 'views/typeGuards/timeRange';
 
+import { TimeRangeDropDownFormValues } from './searchbar/date-time-picker/TimeRangeDropdown';
+
 const formatDatetime = (datetime) => datetime.toString(DateTime.Formats.TIMESTAMP);
 
+export const onSettingSearchBarFormTimeRange = (nextTimeRange: TimeRangeDropDownFormValues['nextTimeRange']) => {
```

Comment:
The naming implies that this is just an event handler, but I think what it is actually doing is normalizing a time range by converting the "all messages" time range. If my impression is correct, I would suggest moving this to a separate module and rename it to something like `normalizeIfAllMessagesRange`.